src/opensignals/data/provider.py

```diff
@@ -0,0 +1,257 @@
+        logger.info(f'Retrieving data for {num} tickers from the database')
+
+        return ticker_data
+
+    @staticmethod
+    def get_ticker_missing(ticker_data,
+                           ticker_map,
+                           last_friday=datetime.today() - relativedelta(weekday=FR(-1))):
+        tickers_available_data = ticker_data.groupby('bloomberg_ticker').agg({'date': [max, min]})
+        tickers_available_data.columns = ['date_max', 'date_min']
+
+        eligible_tickers_available_data = ticker_map.merge(
+            tickers_available_data.reset_index(),
+            on='bloomberg_ticker',
+            how='left'
+        )
+
+        ticker_not_found = eligible_tickers_available_data.loc[
+            eligible_tickers_available_data.date_max.isna(), ['bloomberg_ticker', 'yahoo']
+        ]
+
+        ticker_not_found['start'] = '2002-12-01'
+
+        last_friday_52 = last_friday - relativedelta(weeks=52)
+        tickers_outdated = eligible_tickers_available_data.loc[
+            (
+                    (eligible_tickers_available_data.date_max < last_friday.strftime('%Y-%m-%d')) &
+                    (eligible_tickers_available_data.date_max > last_friday_52.strftime('%Y-%m-%d'))
+            ),
+            ['bloomberg_ticker', 'yahoo', 'date_max']
+        ]
+
+        tickers_outdated['start'] = (
+                tickers_outdated['date_max'] + pd.DateOffset(1)
+        ).dt.strftime('%Y-%m-%d')
+        tickers_outdated.drop(columns=['date_max'], inplace=True)
+
+        return pd.concat(
+            [ticker_not_found, tickers_outdated]
+        )
+
+    @staticmethod
+    def get_live_data(ticker_data,  last_friday):
+        date_string = last_friday.strftime('%Y-%m-%d')
+        live_data = ticker_data[ticker_data.date == date_string].copy()
+
+        # get data from the day before, for markets that were closed
+        last_thursday = last_friday - relativedelta(days=1)
+        thursday_date_string = last_thursday.strftime('%Y-%m-%d')
+        thursday_data = ticker_data[ticker_data.date == thursday_date_string]
+
+        # Only select tickers than aren't already present in live_data
+        thursday_data = thursday_data[
+            ~thursday_data.bloomberg_ticker.isin(live_data.bloomberg_ticker.values)
+        ].copy()
+
+        live_data = pd.concat([live_data, thursday_data])
+        live_data = live_data.set_index('date')
+        return live_data
+
+    @staticmethod
+    def get_train_test_data(ticker_data, feature_names, targets):
+        """merge our feature data with Numerai targets"""
+        ml_data = pd.merge(
+            ticker_data, targets,
+            on=['date', 'bloomberg_ticker'],
+            how='left'
+        )
+
+        logger.info(f'Found {ml_data.target.isna().sum()}'
+                    'rows without target, filling with 0.5')
+        ml_data['target'] = ml_data['target'].fillna(0.5)
+
+        # convert date to datetime and index on it
```

Comment:
comment doesn't apply, no conversion happening here

Reply:
Let's remove it indeed